### PR TITLE
Dynamic load SQL parser test cases from MySQL and PostgreSQL

### LINKED ISSUE #16383: Dynamic load SQL parser test cases from MySQL and PostgreSQL
The current SQL parser test case for unsupported SQL is `org.apache.shardingsphere.test.sql.parser.parameterized.engine.UnsupportedSQLParserParameterizedTest`. 
The test data is in `src/main/resources/sql/unsupported/unsupported.xml` which composed by static and dynamic.

image

We need to keep the static test cases in `UnsupportedSQLParserParameterizedTest` and create a new class, which name maybe `DynamicLoadingSQLParserParameterizedTest`.

The class should load SQL cases from official test cases from MySQL[1] and PostgreSQL[2]. 
The step of `DynamicLoadingSQLParserParameterizedTest`:

1. Load official test cases from GitHub
2. Generate to arrays by parameterized test
3. Run `SQLParserEngine.parse()` try to parse SQL
4. Assert result as `SQLStatement`
5. Catch `SQLParserException` for unsupported SQL
6. Catch `CastException` for parse error
7. Record and print exception number, the final objective is zero exception, please do not assert fail if exception number is not zero, just let the test cases can be done.
8. Please define the test case as IT, it is better to run it in GitHub Action only, the normal `mvn install` should not depend on network environment.

[1] https://github.com/mysql/mysql-server/tree/8.0/mysql-test/t
[2] https://github.com/postgres/postgres/tree/master/src/test/regress/sql

### PR DESCRIPTION
**Pull request by Jacob953**
Fixes #16383.

Changes proposed in this pull request:
  - Accessing files using jsonpath
  - Remove duplicate test cases that can be automatically overwritten

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [x] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `mvn clean install -B -T2C -DskipTests -Dmaven.javadoc.skip=true -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: test/parser/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/engine/DynamicLoadingSQLParserParameterizedTest.java [hunk 1]
```diff
@@ -61,16 +64,11 @@ private static Collection<Map<String, String>> getResponse(final String sqlCaseU
...
-        String nameItem = null;
-        for (String each : lines) {
-            if (each.contains("name")) {
-                nameItem = each.split("\"")[3];
-            } else if (each.contains("download_url")) {
-                String downloadURLItem = each.split("\"")[3];
-                result.add(ImmutableMap.of("name", nameItem, "download_url", downloadURLItem));
-            }
-        }
+        String casesGitHubApoContent = getContent(casesGitHubApiURL);
```
Comment (terrymanu) [inline test/parser/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/engine/DynamicLoadingSQLParserParameterizedTest.java:67]: what's mean of `apo` ?

**Review by terrymanu — CHANGES_REQUESTED** [review]
The unsupported.xml includes lots cases from databases.
Can not remove it directly.
Please pick up and remove the duplicated SQL cases only.

File: test/parser/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/engine/DynamicLoadingSQLParserParameterizedTest.java (hunk 1 above)
Comment (Jacob953) [reply test/parser/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/engine/DynamicLoadingSQLParserParameterizedTest.java:67]: Sorry about that. It should be `Api`.

File: pom.xml [hunk 2]
```diff
@@ -726,6 +727,12 @@
                 <artifactId>caffeine</artifactId>
                 <version>${caffeine.version}</version>
             </dependency>
+            
+            <dependency>
+                <groupId>com.jayway.jsonpath</groupId>
+                <artifactId>json-path</artifactId>
+                <version>${json-path.version}</version>
```
Comment (terrymanu) [inline pom.xml:734]: Please add test scope

Comment (Jacob953) [reply pom.xml:734]: fixed

File: test/parser/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/engine/DynamicLoadingSQLParserParameterizedTest.java (hunk 1 above)
Comment (Jacob953) [reply test/parser/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/engine/DynamicLoadingSQLParserParameterizedTest.java]: fixed

**Review by terrymanu — APPROVED** [review]

**Pull request merged**